main.go

```diff
@@ -142,7 +144,32 @@ func fetchMetricFamilies(url string, ch chan<- *dto.MetricFamily) {
 	if resp.StatusCode != http.StatusOK {
 		log.Fatalf("GET request for URL %q returned HTTP status %s", url, resp.Status)
 	}
+	contentCheck(resp, ch)
+}
 
+func fetchMetricFamiliesWithAuth(url string, ch chan<- *dto.MetricFamily, certificate string, key string) {
+	defer close(ch)
+	cert, err := tls.LoadX509KeyPair(certificate, key)
+	if err != nil {
+		log.Fatal(err)
+	}
+	tlsConfig := &tls.Config{
+		Certificates: []tls.Certificate{cert},
+	}
+	tlsConfig.BuildNameToCertificate()
+	transport := &http.Transport{TLSClientConfig: tlsConfig}
+	client := &http.Client{Transport: transport}
+	resp, err := client.Get(url)
+	if err != nil {
+		log.Fatalf("executing GET request for URL %q failed: %s", url, err)
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		log.Fatalf("GET request for URL %q returned HTTP status %s", url, resp.Status)
+	}
+	contentCheck(resp, ch)
+}
+func contentCheck(resp *http.Response, ch chan<- *dto.MetricFamily) {
```

Comment:
For consistency, use a blank line prior to the function definition.

A better name for this function is probably `decodeContent`.

But my suggestion is anyway to only have the HTTP client creation separately and use the same code for both retrievals otherwise.